src/server/main_service.cc

```diff
@@ -696,29 +723,45 @@ void Service::DispatchCommand(CmdArgList args, facade::ConnectionContext* cntx)
 
   dfly_cntx->cid = cid;
 
-  try {
-    cid->Invoke(args, dfly_cntx);
-  } catch (std::exception& e) {
-    LOG(ERROR) << "Internal error, system probably unstable " << e.what();
+  // Collect stats for all regular transactions and all multi transactions from scripts, except EVAL
+  // itself. EXEC does not use DispatchCommand for dispatching.
+  bool collect_stats =
```

Comment:
I do not understand the second part of condition.
if exec does not use DispatchCommand, then when` (!dfly_cntx->transaction->IsMulti() || under_script)` is false?

Reply:
I wrote it in the comments: except EVAL itself